device/sycl/src/finding/finding_algorithm.sycl

```diff
@@ -0,0 +1,158 @@
+/** TRACCC library, part of the ACTS project (R&D line)
+ *
+ * (c) 2023 CERN for the benefit of the ACTS project
+ *
+ * Mozilla Public License Version 2.0
+ */
+
+// Project include(s).
+#include "../utils/get_queue.hpp"
+#include "traccc/definitions/primitives.hpp"
+#include "traccc/edm/device/finding_global_counter.hpp"
+#include "traccc/finding/candidate_link.hpp"
+#include "traccc/finding/device/apply_interaction.hpp"
+#include "traccc/finding/device/build_tracks.hpp"
+#include "traccc/finding/device/count_threads.hpp"
+#include "traccc/finding/device/find_tracks.hpp"
+#include "traccc/finding/device/make_barcode_sequence.hpp"
+#include "traccc/finding/device/propagate_to_next_surface.hpp"
+#include "traccc/sycl/finding/finding_algorithm.hpp"
+
+// detray include(s).
+#include "detray/core/detector.hpp"
+#include "detray/detectors/telescope_metadata.hpp"
+#include "detray/detectors/toy_metadata.hpp"
+#include "detray/masks/unbounded.hpp"
+#include "detray/propagator/navigator.hpp"
+#include "detray/propagator/rk_stepper.hpp"
+
+// VecMem include(s).
+#include <vecmem/containers/data/vector_buffer.hpp>
+#include <vecmem/containers/device_vector.hpp>
+#include <vecmem/containers/jagged_device_vector.hpp>
+#include <vecmem/containers/vector.hpp>
+
+// DPL include(s).
+#include <oneapi/dpl/algorithm>
+#include <oneapi/dpl/execution>
```

Comment:
And this header (`#include <oneapi/dpl/execution>`) complains with the following error message:
```
In file included from /home/beomki/projects/traccc/traccc/device/sycl/src/finding/finding_algorithm.sycl:37:
In file included from /home/beomki/projects/traccc/traccc_sycl/_deps/dpl-src/include/oneapi/dpl/execution:23:
In file included from /opt/rh/gcc-toolset-10/root/usr/lib/gcc/x86_64-redhat-linux/10/../../../../include/c++/10/execution:32:
In file included from /opt/rh/gcc-toolset-10/root/usr/lib/gcc/x86_64-redhat-linux/10/../../../../include/c++/10/pstl/glue_execution_defs.h:50:
In file included from /opt/rh/gcc-toolset-10/root/usr/lib/gcc/x86_64-redhat-linux/10/../../../../include/c++/10/pstl/algorithm_impl.h:22:
In file included from /opt/rh/gcc-toolset-10/root/usr/lib/gcc/x86_64-redhat-linux/10/../../../../include/c++/10/pstl/parallel_backend.h:16:
/opt/rh/gcc-toolset-10/root/usr/lib/gcc/x86_64-redhat-linux/10/../../../../include/c++/10/pstl/parallel_backend_tbb.h:70:10: error: no member named 'task' in namespace 'tbb'
    tbb::task::self().group()->cancel_group_execution();
    ~~~~~^
/opt/rh/gcc-toolset-10/root/usr/lib/gcc/x86_64-redhat-linux/10/../../../../include/c++/10/pstl/parallel_backend_tbb.h:413:34: error: expected class name
class __merge_task : public tbb::task
                                 ^
/opt/rh/gcc-toolset-10/root/usr/lib/gcc/x86_64-redhat-linux/10/../../../../include/c++/10/pstl/parallel_backend_tbb.h:415:23: error: no type named 'task' in namespace 'tbb'
    /*override*/ tbb::task*
                 ~~~~~^
/opt/rh/gcc-toolset-10/root/usr/lib/gcc/x86_64-redhat-linux/10/../../../../include/c++/10/pstl/parallel_backend_tbb.h:438:6: error: no type named 'task' in namespace 'tbb'
tbb::task*
```